httpx/_content_streams.py

```diff
@@ -328,7 +328,8 @@ def _iter_fields(
             else:
                 yield self.DataField(name=name, value=value)
 
-        for name, value in files.items():
+        file_items = files.items() if isinstance(files, dict) else files
+        for name, value in file_items:  # type: ignore
```

Comment:
No idea how to fix the mypy error on this line `httpx/_content_streams.py:332: error: Unpacking a string is disallowed
`

Reply:
I think it’s because the type hints is using Mapping but we’re only matching against dict (a specific type of mapping), so mypy assumes that the other branch could also receive other types of mappings, and iterating over a mapping yields its keys (strings).

It’s actually possible to `isinstance(obj, typing.Mapping)` (since it’s the same Mapping class than in collections.abc), so I think we can go for that?

Alternatively, switch the `if` to match a tuple (and treat the default case as mapping) - probably a bit less surprising.

Reply:
fixed !